sweep/txgenerator.go

```diff
@@ -169,14 +169,16 @@ func createSweepTx(inputs []input.Input, outputPkScript []byte,
 	}
 
 	// Sweep as much possible, after subtracting txn fees.
-	sweepAmt := int64(totalSum - txFee)
+	sweepAmt := totalSum - txFee
 
 	// The txn will sweep the amount after fees to the pkscript generated
 	// above.
-	sweepTx.AddTxOut(&wire.TxOut{
-		PkScript: outputPkScript,
-		Value:    sweepAmt,
-	})
+	if sweepAmt >= dustLimit {
```

Comment:
should this be an error?

Reply:
If there are no outputs on the tx, it will be caught by the call to `blockchain.CheckTransactionSanity` below.

Reply:
maybe we should call `CheckTransactionSanity` inside this method? idk if we can assume every caller will know it needs to do that check manually.

Reply:
scratch that i see that it already is :)